xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/chaining/AbstractChainingListener.java

```diff
@@ -82,11 +106,11 @@ public AbstractChainingListener()
 
             // Enable the retro compatibility flag if #beginListItem() is overwritten but not #beginListItem(Map)
             if (!withParameter && withoutParameter) {
-                this.beginListItemRetroCompatibility = true;
-
-                break;
+                return true;
```

Comment:
> I've checked and we haven't activated this check

Seems we did actually: https://github.com/xwiki/xwiki-commons/blob/master/xwiki-commons-tools/xwiki-commons-tool-verification-resources/src/main/resources/checkstyle.xml#L329.

Reply:
hmm indeed, I checked, don't know why my search didn't yield results...

Now would need to check why we put 5 instead of the default 3. In any case we'll need to sync since it's 3 on sonarcloud.

Reply:
I see that sergiu did this in 2008:

```
-    <module name="ReturnCount"/>
+    <!-- Increasing the maximum number of allowed return statements, as
otherwise
+         the code will simply use workarounds to hide the same amount of exit
points. -->
+    <module name="ReturnCount">
+      <property name="max" value="5"/>
+    </module>
```